### PR TITLE
issue 1224 added a #focused? method

### PR DESCRIPTION
**Pull request by estensland**
For Issue #1224 

I realized there are requests for aliases on the issue, this is essentially a question on the logic, then I will add aliases and other specs that we can think of, if the core of it makes sense.

I added the focused? method as a chain through focus modules just like the current #focus method, does this make sense?

### CODE REVIEW

**Comment by jasonrclark** [comment]
This looks good to me, thanks! (...and TIL that focused has alternate spellings outside the US. 📚 ✨)

@estensland feel free to add the aliases on if you'd like. I believe I also sent you an invite to team shoes and you'd be able to [merge on your own](https://github.com/shoes/shoes4/wiki/Merging-a-Pull-Request) if you're interested. If you'd rather not, I'm happy to click the big green button.

**Comment by estensland** [comment]
@jasonrclark thanks I appreciate it!

I added some aliases and specs, I'd rather you take a look before merging to make sure I didn't miss an obvious one, and if you think the specs are worth it.

NOTE: I did not do #has_focus? because rubocop disapproved

File: shoes-swt/lib/shoes/swt/common/focus.rb [hunk 1]
```diff
@@ -6,6 +6,14 @@ module Focus
...
           @real.set_focus
         end
+
+        def focused?
+          @real.has_focus
+        end
+
+        alias focussed? focused?
+        alias focussed focused?
+        alias focused focused?
```
Comment (jasonrclark) [inline shoes-swt/lib/shoes/swt/common/focus.rb]: Looking it over, I'd actually be inclined to only provide the aliases on the shoes-core side, not on the GUI. I'm fine with the contract of the backends (which end-users never access directly) saying "you've got to use `focused?` as the primary one, while providing the convenience for Shoes apps themselves of the aliases. Mind making that change and then I'll get it merged?

Comment (estensland) [reply shoes-swt/lib/shoes/swt/common/focus.rb]: Ah that makes sense, will change. Thanks!

Comment (estensland) [reply shoes-swt/lib/shoes/swt/common/focus.rb]: The specs are for the gui and I don't see a spec for the shoes-core. Are the alias specs worth it enough?

Comment (jasonrclark) [reply shoes-swt/lib/shoes/swt/common/focus.rb]: I'm fine with letting this stand if you're ready to be done with the issue.

That said, I've been slowly adding in specs for the various `common` modules that didn't get direct testing way back when they were created (or are tested indirectly through other element types). An example is `shoes-core/spec/shoes/common/visibility_spec.rb`.

I do think it's probably worthwhile to add at the shoes-core level, just to document the expected interaction between shoes-core and the backends, but it's not a huge problem to leave it like it is for now.

Comment (estensland) [reply shoes-swt/lib/shoes/swt/common/focus.rb]: I'll give it a shot!

**Comment by estensland** [comment]
I based the specs off visibility_spec, but let me know if a different formatting or style would be better

File: shoes-core/spec/shoes/common/focus_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  describe "#focused?" do
+    it 'must call #focused? on gui' do
+      expect(gui).to receive(:focused?)
+      subject.focused?
+    end
+
+    it "has alias #focused" do
+      expect(subject.method(:focused) == subject.method(:focused?)).to be true
```
Comment (jasonrclark) [inline shoes-core/spec/shoes/common/focus_spec.rb]: Could these be written more simply as `expect(subject.method(:focused)).to eq(subject.method(:focused?))`?

**Comment by jasonrclark** [comment]
One last small comment, and I've re-kicked the Travis build. There's some intermittent test failure still happening that I haven't managed to track down, so the failure's nothing to do with your code.

**Comment by jasonrclark** [comment]
Woot! Ready to ✨

**Pull request merged**